### PR TITLE
 Bugfix: Inonsistent Brew & Steam switch behaviours #413 #414

### PR DESCRIPTION
**Pull request by genkigenki**
Brew button was not debounced, leading to erratic behaviour if your switches are not perfect.
5s delay removed, as its not needed (was likely introduced due to erratic behaviour due to bouncing switches)
Fixes https://github.com/rancilio-pid/clevercoffee/issues/413

Steam button configured as push button was twice checked and once not debounced. Removed check in main.cpp. Also migrated missing code to switchvoid file from main to ensure behaviour for switch users stays same.
Fixes https://github.com/rancilio-pid/clevercoffee/issues/416

### CODE REVIEW

File: src/brewvoid.h [hunk 1]
```diff
@@ -23,22 +23,21 @@ void checkbrewswitch() {
...
-            if ((millis() - lastDebounceTimeBrewTrigger) > debounceDelayBrewTrigger) {
+            else if ((millis() - lastDebounceTimeBrewTrigger) > debounceDelayBrewTrigger) {
                 // whatever the reading is at, it's been there for longer than the debounce
                 // delay, so take it as the actual current state:
-
-                // if the button state has changed:
-                if (reading != buttonStateBrewTrigger) {
-                    buttonStateBrewTrigger = reading;
+                
+                if (brewswitchTrigger != reading)
```
Comment (kjyv) [inline src/brewvoid.h:36]: Could you put the closing bracket inline with the if statement? It's the code style we generally use

**Review by kjyv — COMMENTED** [review]
lgtm, although not sure about the 5 second delay. Could this be there for making the (software) brew detection work? The comment mentions that. So please test if it is really unnecessary

File: src/brewvoid.h (hunk 1 above)
Comment (genkigenki) [reply src/brewvoid.h:36]: Can't be. Its the mechanical switch, converted to a push button:

Code is executed if:
#if BREWSWITCHTYPE == 2
        #if (PIN_BREWSWITCH > 0)

Config file:
#define BREWSWITCHTYPE 2           // 0 = no switch connected, 1 = normal switch, 2 = trigger switch

I tested with the brew switch, no issues at all.

Comment (kjyv) [reply src/brewvoid.h:36]: Ah yeah, the code path wouldn't make sense with software brew detection - the trigger switch is detected already. Then it should be fine

**Review by kjyv — APPROVED** [review]

**Review by LoQue90 — CHANGES_REQUESTED** [review]
works worse than befor. dosen´t react do all pushes, brew start/stops not reliable
debounceDelayBrewTrigger set to 80 for test, did not help

**Comment by kjyv** [comment]
Thanks for testing again. My review was not based on actually testing this as I don't have a push button installed in my machine, guess I should have added that caveat

**Comment by LoQue90** [comment]
> Thanks for testing again. My review was not based on actually testing this as I don't have a push button installed in my machine, guess I should have added that caveat

don´t worry, the code is fine and works.
Problem is the debounce timing. Look like my stil of pushing that button is more a snap, which needs a faster debounce of 20ms.
Just wanted to check that :)

**Pull request closed without merging**